include/bluetooth/conn.h

```diff
@@ -386,6 +386,15 @@ struct bt_conn_cb {
  */
 void bt_conn_cb_register(struct bt_conn_cb *cb);
 
+/** Enable/disable bonding.
+ *
+ *  Set/clear the Bonding flag in the Authentication Requirements of
+ *  SMP Pairing Request/Response data. The flag is enabled by default.
```

Comment:
The last sentence is not accurate, is it? The default depends on the Kconfig setting. You should also make it clear that for the vast majority of apps calling this shouldn't be needed. What was the end-product use case that made you decide that this is needed in a wider context than just for testing?

Reply:
@MariuszSkamra I discussed this with @carlescufi on IRC and the consensus was to keep this public. So please just address the two other issues I brought up.